vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/VssDefinitionProcessor.kt

```diff
@@ -128,11 +130,50 @@ class VssDefinitionProcessor(
                     duplicateSpecificationNames,
                 )
 
-                val file = FileSpec.builder(PACKAGE_NAME, classSpec.name!!)
+                val className = classSpec.name ?: throw NoSuchFieldException("")
+                val fileSpecBuilder = FileSpec.builder(PACKAGE_NAME, className)
+
+                val parentImport = createParentImport(specModel, generatedFilesVssPathToClassName)
+                if (parentImport.isNotEmpty()) {
+                    fileSpecBuilder.addImport(PACKAGE_NAME, parentImport)
+                }
+
+                val file = fileSpecBuilder
                     .addType(classSpec)
                     .build()
 
                 file.writeTo(codeGenerator, false)
+                generatedFilesVssPathToClassName[vssPath.path] = className
+            }
+        }
+
+        // Uses a map of vssPaths to ClassNames which are validated if it contains a parent of the given specModel.
+        // If the actual parent is a sub class (Driver) in another class file (e.g. Vehicle) then this method returns
+        // a sub import e.g. "Vehicle.Driver". Otherwise just "Vehicle" is returned.
+        private fun createParentImport(
+            specModel: VssSpecificationSpecModel,
+            parentVssPathToClassName: Map<String, String>,
+        ): String {
+            var availableParentVssPath = specModel.vssPath
+            var parentSpecClassName: String? = null
+
+            // Iterate up from the parent until the actual file name = class name was found. This indicates
+            // that the actual parent is a sub class in this file.
+            while (availableParentVssPath.contains(".")) {
+                availableParentVssPath = availableParentVssPath.substringBeforeLast(".")
+
+                parentSpecClassName = parentVssPathToClassName[availableParentVssPath]
+                if (parentSpecClassName != null) break
+            }
+
+            if (parentSpecClassName == null) return ""
```

Comment:
Optional:
We have implicite knowledge, that only Vehicle should have a null parent. Should we adapt the logic, to throw an Exception or at least log an error when this occurs on any other (unexpected) VssPath?

Reply:
For the method itself both cases are valid. A parent was found or no parent was found (then it returns an empty string). So I don't like putting it here but I will add a log entry. Checking this case will result in an "ugly" if case. I prefer not validating this case. :)